libnetdata/socket/security.c

```diff
@@ -21,35 +21,43 @@ int netdata_validate_server =  NETDATA_SSL_VALID_CERTIFICATE;
  * @param ret the return of the caller
  */
 static void security_info_callback(const SSL *ssl, int where, int ret __maybe_unused) {
-    (void)ssl;
+    UNUSED(ssl);
     if (where & SSL_CB_ALERT) {
-        debug(D_WEB_CLIENT,"SSL INFO CALLBACK %s %s", SSL_alert_type_string(ret), SSL_alert_desc_string_long(ret));
+        debug(D_WEB_CLIENT,"SSL INFO CALLBACK %s", SSL_alert_desc_string_long(ret));
     }
 }
 
 /**
- * OpenSSL Library
+ * Test Certificate
+ *
+ * Check the certificate of Netdata parent
+ *
+ * @param ssl is the connection structure
  *
- * Starts the openssl library for the Netdata.
+ * @return It returns 0 on success and -1 otherwise
  */
-void security_openssl_library()
-{
-#if OPENSSL_VERSION_NUMBER < OPENSSL_VERSION_110
-# if (SSLEAY_VERSION_NUMBER >= OPENSSL_VERSION_097)
-    OPENSSL_config(NULL);
-# endif
-
-    SSL_load_error_strings();
+int security_test_certificate(SSL *ssl) {
+    X509* cert = SSL_get_peer_certificate(ssl);
+    long status;
+    if (!cert) {
+        return -1;
+    }
 
-    SSL_library_init();
-#else
-    if (OPENSSL_init_ssl(OPENSSL_INIT_LOAD_CONFIG, NULL) != 1) {
-        error("SSL library cannot be initialized.");
+    status = SSL_get_verify_result(ssl);
+    if((X509_V_OK != status))
+    {
+        char error[512];
+        ERR_error_string_n(ERR_get_error(), error, sizeof(error));
+        error("SSL RFC4158 check:  We have a invalid certificate, the tests result with %ld and message %s",
+              status, error);
+        return -1;
     }
-#endif
+
+    return 0;
 }
 
-#if OPENSSL_VERSION_NUMBER >= OPENSSL_VERSION_110
+
+#if defined(OPENSSL_VERSION_110) || defined(NETDATA_HTTPS_WITH_WOLFSSL)
```

Comment:
Did you want to check for `OPENSSL_VERSION_NUMBER`? `OPENSSL_VERSION_110` looks ugly.

Reply:
All right, I updated.